lib/msal-node/src/cache/CacheManager.ts

```diff
@@ -0,0 +1,73 @@
+import { Storage } from './Storage';
+import { Serializer, Deserializer, InMemoryCache, JsonCache } from '@azure/msal-common';
+import { ICachePlugin } from './ICachePlugin';
+
+const defaultSerializedCache: JsonCache = {
+    Account: {},
+    IdToken: {},
+    AccessToken: {},
+    RefreshToken: {},
+    AppMetadata: {}
+};
+
+
+export class CacheManager {
+
+    hasChanged: boolean;
+    storage: Storage;
+    persistence: ICachePlugin;
+
+    constructor(storage: Storage) {
+        this.hasChanged = false;
+        this.storage = storage;
+        this.storage.registerChangeEmitter(this.handleChangeEvent.bind(this))
+
+    }
+
+    setPersistence(persistence: ICachePlugin): void {
+        this.persistence = persistence;
+    }
+
+    handleChangeEvent() {
+        this.hasChanged = true;
+    }
+
+    async serialize(): Promise<string> {
+        const cache: string = JSON.stringify(Serializer.serializeAllCache(this.storage.getCache()));
+        if (this.persistence) {
+            await this.persistence.writeToStorage(cache);
```

Comment:
After thinking about this for a bit, I realized that `await this.persistence.writeToStorage(cache);` needs to be triggered from within `Storage`. The library needs the latest state from disk when doing any operation that modifies the cache, not when serializing or deserializing. 

Essentially, if we are going to follow this model, where msal triggers a `await writeToStorage` or `await readFromStorage()`, Storage has to be made async. I'm not sure we can get around that. The first PR you put out, where it's up to the application to read and write to disk after any MSAL operation that modify the cache is the only way that we can keep synchronous. That PR is missing a couple of things (like hasStateChanged) but I think it could work, although it's not nearly as friendly for customers as this model (it relies on people knowing what MSAL operations alter the cache, because they have to wrap those call in disk reads and writes).